python/lbann/launcher/slurm.py

```diff
@@ -1,136 +1,150 @@
-        reservation (str, optional): Scheduler reservation name.
-        srun_args (str, optional): Command-line arguments to srun.
-        environment (dict of {str: str}, optional): Environment
-            variables.
-        setup_only (bool, optional): If true, the experiment is not
-            run after the batch script is created.
-
-    Returns:
-        int: Exit status from Slurm. This is really only meaningful if
-            the script is run on an existing node allocation. If a
-            batch job is submitted, Slurm will probably return 0
-            trivially.
-
-    """
-
-    # Check for an existing job allocation from Slurm
-    # Note: Settings for current job allocation take precedence
-    has_allocation = 'SLURM_JOB_ID' in os.environ
-    if has_allocation:
-        job_name = os.environ['SLURM_JOB_NAME']
-        partition = os.environ['SLURM_JOB_PARTITION']
-        account = os.environ['SLURM_JOB_ACCOUNT']
-        time_limit = -1
-
-    # Experiment directory
-    experiment_dir = os.path.abspath(experiment_dir)
-    os.makedirs(experiment_dir, exist_ok=True)
-    batch_file = os.path.join(experiment_dir, 'batch.sh')
-    out_file = os.path.join(experiment_dir, 'out.log')
-    err_file = os.path.join(experiment_dir, 'err.log')
-    nodes_file = os.path.join(experiment_dir, 'nodes.txt')
-
-    # Write batch script
-    with open(batch_file, 'w') as f:
-        f.write('#!/bin/sh\n')
-
-        # Slurm job settings
+        # Configure header with Slurm job options
+        self._construct_header(job_name=job_name,
+                               nodes=self.nodes,
+                               time_limit=time_limit,
+                               partition=partition,
+                               account=account)
+
+    def _construct_header(self,
+                          job_name=None,
+                          nodes=1,
+                          time_limit=None,
+                          partition=None,
+                          account=None):
+        """Construct script header with options for sbatch."""
         if job_name:
-            f.write('#SBATCH --job-name={}\n'.format(job_name))
-        f.write('#SBATCH --nodes={}\n'.format(nodes))
-        if partition:
-            f.write('#SBATCH --partition={}\n'.format(partition))
-        if account:
-            f.write('#SBATCH --account={}\n'.format(account))
-        if reservation:
-            raise ValueError('Slurm reservations not supported')
-        f.write('#SBATCH --workdir={}\n'.format(experiment_dir))
-        f.write('#SBATCH --output={}\n'.format(out_file))
-        f.write('#SBATCH --error={}\n'.format(err_file))
-        if time_limit >= 0:
+            self.add_header_line('#SBATCH --job-name={}'.format(job_name))
+        self.add_header_line('#SBATCH --nodes={}'.format(nodes))
+        if time_limit is not None:
+            time_limit = max(time_limit, 0)
             seconds = int((time_limit % 1) * 60)
```

Comment:
We specify in the documentation that `time_limit` is an `int`, so this is always `0`.

Reply:
Also, since we document `time_limit` as being in minutes, we shouldn't bother to compute seconds.

Reply:
I think those are valid points, but it doesn't really hurt us to be a bit more robust than our documentation promises. I'm also endlessly paranoid about the user passing in weird arguments.